src/shared/condition.c

```diff
@@ -138,6 +140,49 @@ static int condition_test_kernel_command_line(Condition *c) {
         return false;
 }
 
+static int condition_test_user(Condition *c) {
+        uid_t id;
+        int r;
+        _cleanup_free_ const char * username = NULL;
+
+        assert(c);
+        assert(c->parameter);
+        assert(c->type == CONDITION_USER);
+
+        r = parse_uid(c->parameter, &id);
+        if (r >= 0)
+                return id == getuid() || id == geteuid();
+        else if (r != -EINVAL)
```

Comment:
I wouldn't check for the precise reason here I must say... just use it as username directly, there's no real reason to do that only if the string failed to qualify as UID for a a very specific reason